bcda/cclf/cclf.go

```diff
@@ -153,17 +153,6 @@ func importCCLF8(ctx context.Context, fileMetadata *cclfFileMetadata) error {
 		maxPendingQueries: utils.GetEnvInt("STATEMENT_EXEC_COUNT", 200000),
 	}
 
-	err := importCCLF(ctx, fileMetadata, importer)
-
-	if err != nil {
-		updateImportStatus(fileMetadata, constants.ImportFail)
-		return err
-	}
-	updateImportStatus(fileMetadata, constants.ImportComplete)
-	return nil
-}
-
-func importCCLF(ctx context.Context, fileMetadata *cclfFileMetadata, importer importer) (err error) {
```

Comment:
We should undo the changes that removed the `return err`. If we do not do an early return when we encounter an error, we may run into nil references/other issues. Ex: if `zip.NewReader` returns an error, the reader it returns is nil. [Ref.](https://golang.org/src/archive/zip/reader.go?s=904:953#L39)

I think I understand why the early returns were removed (so we set the import status to failed), but we can achieve it by using `defer`.

If we switch the signature to use a named return value (`func importCCLF8(ctx context.Context, fileMetadata *cclfFileMetadata) (err error)`) and switch any instances where we reallocate an error value (i.e. `err := ...`), we can do the following.

```
defer func() {
		if err != nil {
			updateImportStatus(fileMetadata, constants.ImportFail)
		}
	}()
```

We should place this defer statement after we've guaranteed that `fileMetadata` is not nil.